libcontainer/rootfs_linux.go

```diff
@@ -1299,6 +1299,12 @@ func setRecAttr(m *configs.Mount, rootfs string) error {
 	if m.RecAttr == nil {
 		return nil
 	}
+	// Make sure that we don't have MOUNT_ATTR_IDMAP specified here. id-mapped
+	// mounts are handled through procMountPlease, and we cannot apply this
+	// attribute here. This should never happen.
+	if (m.RecAttr.Attr_set|m.RecAttr.Attr_clr)&unix.MOUNT_ATTR_IDMAP != 0 {
+		return errors.New("[internal error] mount configuration cannot contain recAttr for MOUNT_ATTR_IDMAP")
+	}
```

Comment:
Do we want to validate this on the validator too? :thinking:

Reply:
Good point. I put this here when the way `idmap` and `ridmap` were represented were such that you couldn't do the check in the validator.